### PR TITLE
#8688 additional props for error reporting

### LINKED ISSUE #8688: Include url prop in rum error detail telemetry
Because of how we emit our error details in an offscreen page, we don't capture what page the user was on at the time of the error occurring.

The error log details always includes the same url:
```
url
chrome-extension://mpjjildhmpddojocokjkgmlkkkfjnepo/offscreen.html
```

Ensure we capture the url and any other props like referrer at the time of error reporting.

### PR DESCRIPTION
**Pull request by fungairino**
## What does this PR do?

Closes #8688 

This PR enhances the telemetry logging capabilities of the Pixiebrix extension by adding three new fields to the logging context: `url`, `connectionType`, and `referrer`. These fields are captured in the `reportError` function and added to the `MessageContext` type.

Manually verified the output in the offscreen payload:
```
{
    "service": "pixiebrix-browser-extension",
    "session_id": "9cbcdeaa-ecc4-4e6f-a76d-0add0a673b73",
    "usr": {
        "id": "1701b77a-5ccc-4255-b768-55ee07ceca5e",
        "email": "eduardo+test@pixiebrix.com",
        "organizationId": "7939ffb1-1ac6-49cd-954e-0e3b25d93a95"
    },
    "view": {
        "referrer": "",
        "url": "chrome-extension://mpjjildhmpddojocokjkgmlkkkfjnepo/offscreen.html"
    },
    "code_version": "014fa5d59",
    "date": 1719254853748,
    "message": "Sidebar should not be loaded in an iframe",
    "status": "error",
    "origin": "logger",
    "error": {
        "stack": "Error: Sidebar should not be loaded in an iframe\n  at handleNavigate @ extension://dynamichost/bundles/contentScriptCore.bundle.js:17551:76\n  at async initNavigation @ extension://dynamichost/bundles/contentScriptCore.bundle.js:17607:5",
        "kind": "Error",
        "message": "Sidebar should not be loaded in an iframe"
    },
    "pageName": "contentScript",
    "url": "https://pbx.vercel.app/",
    "connectionType": "4g",
    "referrer": "",
    "extensionVersion": "2.0.4",
    "manifestVersion": 3,
    "runtimeId": "mpjjildhmpddojocokjkgmlkkkfjnepo",
    "name": "Error",
    "stack": "Error: Sidebar should not be loaded in an iframe\n    at handleNavigate (chrome-extension://mpjjildhmpddojocokjkgmlkkkfjnepo/bundles/contentScriptCore.bundle.js:17551:76)\n    at async initNavigation (chrome-extension://mpjjildhmpddojocokjkgmlkkkfjnepo/bundles/contentScriptCore.bundle.js:17607:5)"
}
```

## Checklist

- [ ] Add jest or playwright tests and/or storybook stories - Pending unit testing

### CODE REVIEW

File: src/telemetry/reportError.ts [hunk 1]
```diff
@@ -65,6 +65,15 @@ export default function reportError(
       ...context,
       // Add on the reporter side of the message. On the receiving side it would always be `background`
       pageName: getContextName(),
+      // Record original current url and referrer here before it is lost in the service worker.
+      url: window?.location.href,
+      referrer: document?.referrer,
+      // Network speed. "4g", "3g", "2g", "slow-2g" https://developer.mozilla.org/en-US/docs/Glossary/Effective_connection_type
+      connectionType:
+        // Casting effectiveType since isn't currently on the navigator type
+        // TODO: remove this cast when the TS lib type for navigator is updated
```
Comment (mnholtz) [inline src/telemetry/reportError.ts:74]: Is there a ticket for this anywhere? So the next person that comes across this comment can just do a quick check

Comment (fungairino) [reply src/telemetry/reportError.ts:74]: No. It appears that this is is an experimental property that's currently only supported in chromium. I can switch this from a todo to just a link to this:
https://github.com/microsoft/TypeScript/issues/56962

Comment (mnholtz) [reply src/telemetry/reportError.ts:74]: Yeah, let's do that

File: src/telemetry/reportError.test.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  it("should not log error to console if logToConsole is false", () => {
+    const error = new Error("Test error");
+    reportError(error, { logToConsole: false });
+    expect(consoleErrorSpy).not.toHaveBeenCalled();
+  });
+
+  it("should not report error if it should be ignored", () => {
+    // "ResizeObserver loop limit exceeded" is an ignored error
+    const error = new Error("ResizeObserver loop limit exceeded");
```
Comment (mnholtz) [inline src/telemetry/reportError.test.ts:58]: nit: (would remove the need for the comment)
```suggestion
    const shouldBeIgnoredError = new Error("ResizeObserver loop limit exceeded");
```

**Review by mnholtz — APPROVED** [review]

**Pull request merged**